strings/check_anagram.nim

```diff
@@ -0,0 +1,173 @@
+## Check Anagram
+## =============
+## wiki: https://en.wikipedia.org/wiki/Anagram
+##
+## Two words are anagrams if:
+## - They are made up of the same letters.
+## - The letters are arranged differently.
+## - They are not the same word (a word is not an anagram of itself).
+## - The case of word's characters is ignored.
+##
+## A word is any string of characters `{'A'..'Z', 'a'..'z'}`. Other characters,
+## including whitespace and puctuation, are considered invalid and raise a
+## `ValueError` exception.
+##
+## Note: Generate full doc with `nim doc --docinternal check_anagram.nim`
+
+runnableExamples:
+  doAssert "coder".isAnagram("credo")
+
+  doAssert not "Nim".isAnagram("Nim")
+
+  doAssert not "parrot".isAnagram("rapport")
+
+  let raisedException = try:
+      discard "Pearl Jam".isAnagram("Maple Jar")
+      false
+    except ValueError:
+      true
+  doassert raisedException
```

Comment:
I think top-level runnableExamples on top are great.
Proc-level RE are hard to distinguish from code but are pretty useful in the documentation of a large module.
Since most the Algorithms files are "snippets", I think we should not enforce proc-level RE, but enforce both a RE (for usage showcase) and a testsuite (for validity).
I will update the README.md accordingly.